### PR TITLE
chore: bump module path to v8

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

closes: #4377

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/go.mod [hunk 1]
```diff
@@ -268,10 +233,12 @@ replace (
...
 // uncomment to use the local version of ibc-go, you will need to run `go mod tidy` in e2e directory.
-replace github.com/cosmos/ibc-go/v7 => ../
+replace github.com/cosmos/ibc-go/v8 => ../
 
 replace github.com/syndtr/goleveldb => github.com/syndtr/goleveldb v1.0.1-0.20210819022825-2ae1ddf74ef7
 
 replace github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
 
-replace github.com/strangelove-ventures/interchaintest/v7 => github.com/notional-labs/interchaintest/v7 v7.0.0-20230831143727-c9539fd05629
+//replace github.com/strangelove-ventures/interchaintest/v8 => github.com/notional-labs/interchaintest/v8 v8.0.0-20230905211440-c2dc240d2221
```
Comment (colin-axner) [inline e2e/go.mod:242]:
```suggestion
```

File: e2e/tests/upgrades/upgrade_test.go [hunk 2]
```diff
@@ -7,29 +7,26 @@ import (
 	"time"
 
 	"github.com/cosmos/gogoproto/proto"
-	intertxtypes "github.com/cosmos/interchain-accounts/x/inter-tx/types"
-	interchaintest "github.com/strangelove-ventures/interchaintest/v7"
-	"github.com/strangelove-ventures/interchaintest/v7/chain/cosmos"
-	"github.com/strangelove-ventures/interchaintest/v7/ibc"
-	test "github.com/strangelove-ventures/interchaintest/v7/testutil"
+	// intertxtypes "github.com/cosmos/interchain-accounts/x/inter-tx/types"
```
Comment (colin-axner) [inline e2e/tests/upgrades/upgrade_test.go:10]:
```suggestion
```

**Comment by damiannolan** [comment]
Opened https://github.com/cosmos/ibc-go/issues/4592 to follow-up on intertx tests.

**Review by colin-axner — APPROVED** [review]
Ay! I reviewed all the files!

**Pull request merged**